fil-proofs-tooling/src/bin/benchy/rational_post.rs

```diff
@@ -0,0 +1,167 @@
+use std::collections::BTreeMap;
+use std::fs::{create_dir_all, OpenOptions};
+use std::io::stdout;
+use std::time::{Duration, Instant};
+
+use cpu_time::ProcessTime;
+use fil_proofs_tooling::metadata::Metadata;
+use filecoin_proofs::fr32::write_padded;
+use filecoin_proofs::pieces::get_aligned_source;
+use filecoin_proofs::types::{
+    PaddedBytesAmount, PoRepConfig, PoRepProofPartitions, PoStConfig, SectorSize,
+    UnpaddedBytesAmount,
+};
+use filecoin_proofs::{generate_post, seal, verify_post, PrivateReplicaInfo, PublicReplicaInfo};
+use log::info;
+use rand::random;
+use storage_proofs::sector::SectorId;
+
+// All files opened and written to will have this file path preface.
+const BASE_DIR: &str = "/var/tmp/benchy-rational-post";
+
+// The seed for the rng used to generate which sectors to challenge.
+const CHALLENGE_SEED: [u8; 32] = [0; 32];
+
+const PROVER_ID: [u8; 31] = [0; 31];
+const SECTOR_ID: u64 = 0;
+const N_PARTITIONS: PoRepProofPartitions = PoRepProofPartitions(1);
+
+#[derive(Serialize)]
+#[serde(rename_all = "kebab-case")]
+struct Inputs {
+    sector_size: usize,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "kebab-case")]
+struct Outputs {
+    proving_cpu_time_ms: u64,
+    proving_wall_time_ms: u64,
+    verifying_wall_time_ms: u64,
+    verifying_cpu_time_ms: u64,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "kebab-case")]
+struct Report {
+    inputs: Inputs,
+    outputs: Outputs,
+}
+
+impl Report {
+    /// Print all results to stdout
+    pub fn print(&self) {
+        let wrapped = Metadata::wrap(&self).expect("failed to retrieve metadata");
+        serde_json::to_writer(stdout(), &wrapped).expect("cannot write report JSON to stdout");
+    }
+}
+
+struct FuncMeasurement<T> {
```

Comment:
BLOCKING: I see that `FuncMeasurement` is also defined in `benchy::zigzag`. Would you mind refactoring such that the definition can be shared between the two modules?